src/platform_impl/windows/event.rs

```diff
@@ -343,6 +343,7 @@ pub fn handle_extended_keys(
     extended: bool,
 ) -> Option<(c_int, UINT)> {
     // Welcome to hell https://blog.molecular-matters.com/2011/09/05/properly-handling-keyboard-input/
+    scancode = if extended { 0xE000 } else { 0x0000 } | scancode;
```

Comment:
This affects the PAUSE/SCROLL logic below as it doesn't filter the additional event caused by RAWINPUT:

PR:
```
DeviceEvent { device_id: DeviceId(DeviceId(65599)), event: Key(KeyboardInput { scancode: 57373, state: Released, virtual_keycode: Some(Pause), modifiers: (empty) }) }
WindowEvent { window_id: WindowId(WindowId(0x75065e)), event: KeyboardInput { device_id: DeviceId(DeviceId(0)), input: KeyboardInput { scancode: 69, state: Released, virtual_keycode: Some(Pause), modifiers: (empty) }, is_synthetic: false } }
DeviceEvent { device_id: DeviceId(DeviceId(65599)), event: Key(KeyboardInput { scancode: 69, state: Released, virtual_keycode: Some(Pause), modifiers: (empty) }) }
```

master branch:
```
WindowEvent { window_id: WindowId(WindowId(0xd06cc)), event: KeyboardInput { device_id:  DeviceId(DeviceId(0)), input: KeyboardInput { scancode: 69, state: Released, virtual_keycode: Some(Pause), modifiers: (empty) }, is_synthetic: false } }
DeviceEvent { device_id: DeviceId(DeviceId(65599)), event: Key(KeyboardInput { scancode: 69, state: Released, virtual_keycode: Some(Pause), modifiers: (empty) }) }
```

Reply:
I've updated this logic. I tested master and found it was reporting the same scancode and virtual key for scancode and numlock. The updated version should handle them both as separate entities better. I also tested scroll lock, pause and numlock with a variety of modifiers and it seems to be looking happy now.